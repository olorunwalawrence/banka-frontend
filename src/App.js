import React from 'react';

import Routes from './routes/appRoutes';

const App = () => (
  (
    <>
      <Routes />
    </>
  )
);

export default App;
