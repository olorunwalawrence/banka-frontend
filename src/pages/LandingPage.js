import React from 'react';
import { Link } from 'react-router-dom';
import { marginRight } from '../components/styles'

export default () => (
  <>
    <header className="grid head">
      <div className="image" />
      <div className="container">
        <h1>
          {/* <strong>M</strong>
          y */}
          <strong>B</strong>
          anka
        </h1>
        <p> We keep our words ....</p>
        <Link to="/login" className="btn btn-green" style={marginRight}>Login</Link>
        <Link to="/signup" className="btn btn-border-y">Signup</Link>
      </div>
    </header>
  </>
);