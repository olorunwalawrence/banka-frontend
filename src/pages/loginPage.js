import React from 'react';
import LoginLayout from '../components/login/loginLayout';

export default () => (
  <LoginLayout />
)
