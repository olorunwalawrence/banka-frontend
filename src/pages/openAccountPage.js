import React from 'react';
import Layout from '../components/openAccount/openAccountLayout';

export default () => (
  <Layout />
)