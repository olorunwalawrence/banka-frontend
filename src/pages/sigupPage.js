import React from 'react';
import SignupLayout from '../components/signup/signupLayout';

const SignupPage = () => (
  <SignupLayout />
)

export default SignupPage;
