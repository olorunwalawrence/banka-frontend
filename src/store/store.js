import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import rootReducer from './reducers';

const mergeMiddlewares = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
const initialState = {};
const middleware = [thunk];

const store = createStore(rootReducer, initialState, mergeMiddlewares(applyMiddleware(...middleware)));

export default store;
