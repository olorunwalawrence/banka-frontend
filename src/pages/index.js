import LandingPage from './LandingPage';
import SignupPage from './sigupPage';
import UserHomepage from './userHomePage';
import LoginPage from './loginPage';
import OpenAccountPage from './openAccountPage';

export { SignupPage, LandingPage, UserHomepage, LoginPage, OpenAccountPage };
