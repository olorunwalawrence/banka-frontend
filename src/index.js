import React from 'react';
import ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import store from './store/store';
import App from './App';
import './public/css/custom.css';
import './public/css/icofonts/icofont.min.css';

ReactDOM.render(
  <Provider store={store}>
    <App />
  </Provider>,
  document.getElementById('app')
);
