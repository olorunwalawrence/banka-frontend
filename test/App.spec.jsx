import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow, mount } from './enzyme';
import { Provider } from 'react-redux';
import store from '../src/store/store';
import { BrowserRouter as Router } from 'react-router-dom';
import App from '../src/App';
import Routes from '../src/routes/appRoutes';
import UserHeader, { Header, drop } from '../src/components/header';
import { LandingPage, SignupPage, UserHomepage, LoginPage, OpenAccountPage } from '../src/pages';

describe('App', () => {
  it('should render the App component correctly', () => {
    const shallowWrapper = shallow(<App />);

    expect(toJson(shallowWrapper)).toMatchSnapshot();
  });
});

describe('Routes', () => {
  it('should render the routes for the application correctly', () => {
    const shallowWrapper = shallow(<Routes />);

    expect(toJson(shallowWrapper)).toMatchSnapshot();
  });
});

describe('Header', () => {
  it('should render the header for the application correctly', () => {
    const shallowWrapper = shallow(<Header />);

    expect(toJson(shallowWrapper)).toMatchSnapshot();
  });
  it('should hide the div with className .drop-menu on button click', () => {
    const wrapper = mount(<Provider store={store}><Router><UserHeader /></Router></Provider>);

    expect(toJson(wrapper)).toMatchSnapshot();
  })
});

describe('Landing page', () => {
  it('shoul render the landing page for the application correctly', () => {
    const shallowWrapper = shallow(<LandingPage />);

    expect(toJson(shallowWrapper)).toMatchSnapshot();
  });
});

describe('Signup page', () => {
  it('shoul render the landing page for the application correctly', () => {
    const shallowWrapper = shallow(<App><SignupPage /></App>);
    expect(toJson(shallowWrapper)).toMatchSnapshot();
  });
  it('should contain the display of the completed component', () => {
    const wrapper = mount(
      <Provider store={store}>
        <Router>
          <SignupPage />
        </Router>
      </Provider>
    );
    
    expect(wrapper.find('SignupLayout')).toBeTruthy();
    expect(wrapper.find('header')).toBeTruthy();
    expect(wrapper.find('.back')).toBeTruthy();
    expect(wrapper.find('.grid')).toBeTruthy();
  })
});

describe('Login page', () => {
  it('shoul render the landing page for the application correctly', () => {
    const shallowWrapper = shallow(<App><LoginPage /></App>);
    expect(toJson(shallowWrapper)).toMatchSnapshot();
  });
  it('should contain the display of the completed component', () => {
    const wrapper = mount(
      <Provider store={store}>
        <Router>
          <LoginPage />
        </Router>
      </Provider>
    );
    
    expect(wrapper.find('SignupLayout')).toBeTruthy();
    expect(wrapper.find('header')).toBeTruthy();
    expect(wrapper.find('.back')).toBeTruthy();
    expect(wrapper.find('.grid')).toBeTruthy();
  })
});

describe('UserHomePage', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(
      <Provider store={store}>
        <Router>
          <UserHomepage />
        </Router>
      </Provider>
    )
  });
  it('should contain a header and h1', () => {
    expect(wrapper.find('UserHeader')).toBeTruthy();
    expect(wrapper.find('Dashboard')).toBeTruthy();
  });
});

describe('OenAccountPage', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(
      <Provider store={store}>
        <Router>
          <OpenAccountPage />
        </Router>
      </Provider>
    )
  });
  it('should contain a header and h1', () => {
    expect(wrapper.find('UserHeader')).toBeTruthy();
    expect(wrapper.find('Layout')).toBeTruthy();
  });
});
