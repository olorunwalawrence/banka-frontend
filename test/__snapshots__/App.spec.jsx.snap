// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render the App component correctly 1`] = `
<Fragment>
  <_default />
</Fragment>
`;

exports[`Header should hide the div with className .drop-menu on button click 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(UserHeader)>
        <UserHeader
          logout={[Function]}
        >
          <div
            className="container"
            id="heed"
          >
            <h1
              className="white"
            >
              <span>
                <button
                  className="drop-butn nn hiden"
                  onMouseDown={[Function]}
                  type="button"
                >
                  <i
                    className="icofont-navigation-menu icofont-2x"
                  />
                </button>
                MyBanka
              </span>
            </h1>
          </div>
          <ul
            className=" drop-menu hide"
            style={
              Object {
                "display": "none",
              }
            }
          >
            <Link
              to="/home"
            >
              <a
                href="/home"
                onClick={[Function]}
              >
                <li>
                  Dashboard
                </li>
              </a>
            </Link>
            <Link
              to="/account"
            >
              <a
                href="/account"
                onClick={[Function]}
              >
                <li>
                  Create Account
                </li>
              </a>
            </Link>
            <Link
              onClick={[Function]}
              to="/"
            >
              <a
                href="/"
                onClick={[Function]}
              >
                <li>
                  Logout
                </li>
              </a>
            </Link>
          </ul>
        </UserHeader>
      </Connect(UserHeader)>
    </Router>
  </BrowserRouter>
</Provider>
`;

exports[`Header should render the header for the application correctly 1`] = `
<Fragment>
  <header
    className="header grid-l"
  >
    <div
      className="container"
    >
      <h1>
        MyBanka
      </h1>
    </div>
  </header>
</Fragment>
`;

exports[`Landing page shoul render the landing page for the application correctly 1`] = `
<Fragment>
  <header
    className="grid head"
  >
    <div
      className="image"
    />
    <div
      className="container"
    >
      <h1>
        <strong>
          B
        </strong>
        anka
      </h1>
      <p>
         We keep our words ....
      </p>
      <Link
        className="btn btn-green"
        style={
          Object {
            "marginRight": "5px",
          }
        }
        to="/login"
      >
        Login
      </Link>
      <Link
        className="btn btn-border-y"
        to="/signup"
      >
        Signup
      </Link>
    </div>
  </header>
</Fragment>
`;

exports[`Login page shoul render the landing page for the application correctly 1`] = `
<Fragment>
  <_default />
</Fragment>
`;

exports[`Routes should render the routes for the application correctly 1`] = `
<BrowserRouter>
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/signup"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/home"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/login"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/account"
    />
  </Switch>
</BrowserRouter>
`;

exports[`Signup page shoul render the landing page for the application correctly 1`] = `
<Fragment>
  <_default />
</Fragment>
`;
